aws/resource_aws_launch_template.go

```diff
@@ -0,0 +1,976 @@
+				ConflictsWith: []string{"name_prefix"},
+				ValidateFunc:  validateLaunchTemplateName,
+			},
+
+			"name_prefix": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateLaunchTemplateName,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 255 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 255 characters", k))
+					}
+					return
+				},
+			},
+
+			"client_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"latest_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"block_device_mappings": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"no_device": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"virtual_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ebs": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"delete_on_termination": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"encrypted": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"iops": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"kms_key_id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"snapshot_id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"volume_size": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"volume_type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"credit_specification": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu_credits": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"disable_api_termination": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ebs_optimized": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"elastic_gpu_specifications": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
This attribute is listed as required in the [SDK](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#ElasticGpuSpecification). 😄